src/main/resources/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote/script.js

```diff
@@ -1,41 +1,62 @@
-Behaviour.specify("span.pipeline-new-node", 'NewNodeConsoleNote', 0, function(e) {
-    if (e.processedNewNodeConsoleNote) {
-        return
-    }
-    e.processedNewNodeConsoleNote = true
-    var label = e.getAttribute('label')
+var enclosings = new Map() // id → enclosingId
+var labels = new Map() // id → label
```

Comment:
I'm not sure if scripts like this are encapsulated by Jenkins beyond standard JavaScript behavior, but having global vars like this seems like a bad idea. I would probably find the `<pre>` element with class `console-output` and attach these objects to that element instead, but I'm not a JavaScript expert, so I'm not sure what the preferred approach is for this kind of thing.

Reply:
I think it'd be easier to simply wrap the whole script in an IIFE
```js
(function() {
    // ...
})();
```
This way those variables are not in the global scope but easily accessible from anywhere in this script.